api/src/main/java/fr/catcore/server/translations/api/ServerTranslationsAPI.java

```diff
@@ -0,0 +1,113 @@
+package fr.catcore.server.translations.api;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import fr.catcore.server.translations.api.resource.language.LanguageMap;
+import fr.catcore.server.translations.api.resource.language.LanguageReader;
+import fr.catcore.server.translations.api.resource.language.ServerLanguageDefinition;
+import fr.catcore.server.translations.api.resource.language.ServerLanguageManager;
+import net.fabricmc.api.ModInitializer;
+import net.fabricmc.loader.api.FabricLoader;
+import net.fabricmc.loader.api.ModContainer;
+import net.fabricmc.loader.api.SemanticVersion;
+import net.fabricmc.loader.api.VersionParsingException;
+import net.minecraft.util.Language;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Locale;
+import java.util.Map;
+import java.util.function.Supplier;
+
+public class ServerTranslationsAPI implements ModInitializer {
+
+    private static final JsonParser PARSER = new JsonParser();
+
+    @Override
+    public void onInitialize() {
+        init();
+        ServerLanguageManager.INSTANCE.registerReloadStartListener(ServerTranslationsAPI::init);
+    }
+
+    private static void init() {
+        try {
+            VanillaAssets assets = VanillaAssets.get();
+            Map<String, ServerLanguageDefinition> languages = loadLanguageDefinitions(assets);
+            for (ServerLanguageDefinition language : languages.values()) {
+                Supplier<LanguageMap> supplier = () -> loadVanillaLanguage(assets, language);
+                ServerLanguageManager.INSTANCE.addTranslations(language, supplier);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static Map<String, ServerLanguageDefinition> loadLanguageDefinitions(VanillaAssets assets) throws IOException {
+        Map<String, ServerLanguageDefinition> languageDefinitions = new HashMap<>();
+
+        try (BufferedReader read = new BufferedReader(new InputStreamReader(assets.openStream("pack.mcmeta")))) {
```

Comment:
Hmm, ideally the API wouldn't have to do any network requests. But I don't know if there's any way around it?

Reply:
It's needed, because else it would always return the english translation instead of the localized one.

Reply:
I'm talking about vanilla translations

Reply:
The reason is because I'm filling blank translations in other languages, if it doesn't have translated strings in the language it will get them from the english one.
See: https://github.com/gegy1000/Server-Translations/pull/1/files#diff-e41ef280807a736336b87a0ee2fb1b04R170-R178

Reply:
Alternatively can we just package a list of languages & their definitions within the api?

Reply:
then we would need to update it for each minecraft version :thinking:

Reply:
Or,
I can blacklist vanilla keys from being filled :thinking:

Reply:
I think it might be a better compromise than having the HTTP request in the API. Alternatively, we could initialize every language definition with default settings. We don't use the name- right to left would be broken though. :/ I don't know if we could detect that from the language file